File: ModelRelief/Scripts/Api/V1/Interfaces/IFileModel.ts

// ------------------------------------------------------------------------//
// ModelRelief                                                             //
//                                                                         //
// Copyright (c) <2017-2018> Steve Knipmeyer                               //
// ------------------------------------------------------------------------//
"use strict";

import {IModel} from "Scripts/Api/V1/Interfaces/IModel";

/**
 *  Common interface for all file-backed DTOs (e.g. DepthBuffer, Mesh, Model3d).
 *  Not exposed in UX; API only.
 *  @interface
 */
export interface IFileModel extends IModel {

    fileTimeStamp?: Date;
}

File: ModelRelief/Features/Dto/IFileModel.cs
// -----------------------------------------------------------------------
// <copyright file="IFileModel.cs" company="ModelRelief">
// MIT License (https://github.com/steveknipmeyer/ModelRelief/blob/main/MIT-LICENSE.txt)
// </copyright>
// -----------------------------------------------------------------------

namespace ModelRelief.Dto
{
    using System;
    using Microsoft.AspNetCore.Http;

    /// <summary>
    /// Common interface for all DTO file-backed models (e.g. Model3d, Mesh, DepthBuffer).
    /// </summary>
    public interface IFileModel : IModel
    {
        DateTime? FileTimeStamp { get; set; }        // time stamp of backing file
        public IFormFile FormFile { get; set; }
    }
}
